### PR TITLE
feat: Add support for MergeToCell API

### PR DESCRIPTION
**Pull request by ron-gal**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)
- [x] Rollback plan is reviewed and LGTMed
- [x] All new data plane features have a completed end to end testing plan

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/models/MutationTest.java [hunk 1]
```diff
@@ -281,7 +302,12 @@ public void fromProtoTest() {
             ByteString.copyFromUtf8("fake-value"))
         .deleteCells("fake-family", ByteString.copyFromUtf8("fake-qualifier"))
         .deleteFamily("fake-family2")
-        .addToCell("agg-family", "qual1", 1000, 1234);
+        .addToCell("agg-family", "qual1", 1000, 1234)
+        .mergeToCell(
+            "agg-family",
+            "qual2",
+            1000,
+            Base64.getEncoder().encode(BigInteger.valueOf(1234).toByteArray()));
```
Comment (steveniemitz) [inline google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/models/MutationTest.java:310]: can we make this the "real" way instead? I worry about people looking at these tests for examples and doing it wrong

File: samples/snippets/src/main/java/com/example/bigtable/WriteAggregate.java [hunk 2]
```diff
@@ -53,6 +54,35 @@ public static void writeAggregate(String projectId, String instanceId, String ta
...
+      // Bucket the views for an hour into a single count, giving us an hourly view count for a
+      // given page.
+      Instant hourlyBucket = viewTimestamp.truncatedTo(ChronoUnit.HOURS);
+      long hourlyBucketMicros = hourlyBucket.toEpochMilli() * MICROS_PER_MILLI;
+
+      RowMutation rowMutation = RowMutation.create(tableId, rowKey)
+                                  .mergeToCell(COUNT_COLUMN_FAMILY_NAME, 
+                                    "views",
+                                    hourlyBucketMicros, 
+                                    Base64.getEncoder().encode(BigInteger.valueOf(1).toByteArray()));
```
Comment (steveniemitz) [inline samples/snippets/src/main/java/com/example/bigtable/WriteAggregate.java:77]: same here, this wouldn't actually work in real life

Comment (ron-gal) [reply samples/snippets/src/main/java/com/example/bigtable/WriteAggregate.java]: Done

File: google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/models/MutationTest.java (hunk 1 above)
Comment (ron-gal) [reply google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/models/MutationTest.java]: Done

**Review by steveniemitz — APPROVED** [review]

**Pull request merged**